[PSFP1-0004][PFSP1-0008] Profile Landing Screen, Profile Detail

SetyaG opened the pull request:
Closes #23 
Closes #29
***
## Description
Create the profile landing page and edit profile page, navigate from profile landing page to edit profile page

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [TBD]()
Figma | [link](https://www.figma.com/file/YWJimHMl9fq3v4dQBZSuQo/Hi-Fi-User?node-id=388%3A2)
Issue | [link](https://github.com/inxudianz/Project-Famili/issues/23)

## Todos
- [X] UI
- [X] Coordinator
- [ ] Network
- [ ] Tests
- [ ] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.

### UI Changes
UI related changes will be shown here (can be either screenshot or video format)

### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
The **text fields** and **button** has not changed to custom text field and button

SetyaG committed:
Added files for Profile Landing page
- Coordinator, Coordinator Protocol
- Profile Landing View, View controller, View Model, custom Table View Cell

Added assets for Profile Landing page's table view cells

SetyaG committed:
Added Edit Profile page
- Controller
- Xib file
- View Model
- Protocol
Modified Profile Landing Controller to be able to navigate to edit profile

Added Edit Profile Coordinator and its Protocol
Modified Profile Landing Coordinator and its Protocol's function name

Deleted a comment on ProfileLandingController.swift

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/EditProfileCoordinator.swift:
```diff
@@ -0,0 +1,43 @@
+        vm.view = vc
+        vc.viewModel = vm
+        navigationController?.pushViewController(vc, animated: false)
+    }
```
variable first then function please

SetyaG replied:
done

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/EditProfileCoordinator.swift:
```diff
@@ -0,0 +1,43 @@
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import Foundation
```
no need to import foundation

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/ProfileLandingCoordinator.swift:
```diff
@@ -0,0 +1,47 @@
+        vm.view = vc
+        vc.viewModel = vm
+        navigationController?.pushViewController(vc, animated: false)
+    }
```
variable first then functions

SetyaG replied:
done

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/ProfileLandingCoordinator.swift:
```diff
@@ -0,0 +1,47 @@
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import Foundation
```
remove extra import

SetyaG replied:
done

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/ProfileLandingCoordinator.swift:
```diff
@@ -0,0 +1,47 @@
+        vc.navigationItem.configure()
+        vc.navigationController?.navigationBar.prefersLargeTitles = true
+        vc.navigationItem.largeTitleDisplayMode = .always
+        vc.navigationItem.title = "Welcome, User!"
```
Kindly create constant swift file containing String literal or any kind of literal. example can be viewed in sample or component

SetyaG replied:
Created constant

inxudianz commented on MainApp/MainApp/Feature/Profile/CustomTableViewCell.swift:
```diff
@@ -0,0 +1,40 @@
+        cellLabel?.text = text
+    }
+    
+//    override func awakeFromNib() {
+//        super.awakeFromNib()
+//        // Initialization code
+//        setup()
+//    }
+//
+//    func setup() {
+//        let view = loadViewFromNib()
+//        view.frame = bounds
+//        addSubview(view)
+//    }
+//
+//    func loadViewFromNib() -> UIView {
+//        let bundle = Bundle(for: type(of: self))
+//        let nib = UINib(nibName: "CustomTableViewCell", bundle: bundle)
+//        let view = nib.instantiate(withOwner: self, options: nil)[0] as! UIView
+//        return view
+//    }
```
remove dead code

inxudianz commented on MainApp/MainApp/Feature/Profile/CustomTableViewCell.swift:
```diff
@@ -0,0 +1,40 @@
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import Foundation
```
remove foundation

inxudianz commented on MainApp/MainApp/Feature/Profile/EditProfileController.swift:
```diff
@@ -0,0 +1,27 @@
+//  Copyright © 2020 William Inx. All rights reserved.
+//
+
+import Foundation
```
remove foundation

inxudianz commented on MainApp/MainApp/Feature/Profile/EditProfileController.swift:
```diff
@@ -0,0 +1,27 @@
+import Component
+
+class EditProfileController: MasterViewController, EditProfileProtocol {
+    var viewModel: EditProfileViewModelProtocol?
```
outlet first then variable

inxudianz commented on MainApp/MainApp/Feature/Profile/EditProfileViewModel.swift:
```diff
@@ -0,0 +1,22 @@
+        coordinator?.saveEditProfile()
+    }
+    
+    
```
remove extra white spaces

inxudianz commented on MainApp/MainApp/Feature/Profile/ProfileLandingController.swift:
```diff
@@ -0,0 +1,95 @@
+
+class ProfileLandingController: MasterViewController, ProfileLandingViewProtocol, UITableViewDelegate, UITableViewDataSource {
+    
+    var viewModel: ProfileLandingViewModelProtocol?
```
outlet, variable, function

inxudianz commented on MainApp/MainApp/Feature/Profile/ProfileLandingController.swift:
```diff
@@ -0,0 +1,95 @@
+import UIKit
+import Component
+
+class ProfileLandingController: MasterViewController, ProfileLandingViewProtocol, UITableViewDelegate, UITableViewDataSource {
```
refactor tableviewDelegate and datasource to its own file, and get the data from viewModel

inxudianz commented on MainApp/MainApp/Feature/Profile/ProfileLandingController.swift:
```diff
@@ -0,0 +1,95 @@
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Account"
+        case 1:
+            return "General"
+        default:
+            return ""
```
create constant file like above

SetyaG replied:
Created constant file

inxudianz commented on MainApp/MainApp/Feature/Profile/ProfileLandingViewModel.swift:
```diff
@@ -0,0 +1,27 @@
+    
+    func navigateToFreqAskedQuestion() {
+        // Call coordinator to FAQ
+    }
```
variable - > function

SetyaG replied:
Done

inxudianz commented on MainApp/MainApp/Coordinator/Feature/Profile/EditProfileCoordinator.swift:
```diff
@@ -0,0 +1,43 @@
+import Foundation
+import UIKit
+
+class EditProfileCoordinator: EditProfileCoordinatorProtocol {
```
please combine editProfileCoordinator and profileLandingCoordinator as it is one same feature. name it ProfileCoordinator

SetyaG replied:
Combined the coordinator and protocol

inxudianz commented on MainApp/MainApp.xcodeproj/project.pbxproj:
```diff
@@ -313,14 +343,28 @@
+				1EBF012B24D0A6A00098C24A /* ProfileLandingViewModel.swift */,
+				1EBF012D24D0A6B00098C24A /* ProfileLandingProtocol.swift */,
+				1EBF014924D1E5450098C24A /* CustomTableViewCell.swift */,
+				1EBF014B24D1E5510098C24A /* CustomTableViewCell.xib */,
```
Please seperate custom cell into its own group

SetyaG replied:
Moved to cell to its own folder

inxudianz commented on MainApp/MainApp.xcodeproj/project.pbxproj:
```diff
@@ -313,14 +343,28 @@
 				923249AB24C0406B00FDFC3E /* Resources */,
+				1E44DC5124D855C600025AE8 /* EditProfileController.swift */,
+				1E44DC5524D8560400025AE8 /* EditProfileController.xib */,
+				1E44DC5724D8562D00025AE8 /* EditProfileProtocol.swift */,
```
please create group inside feature/profile called edit and put the edit related stuff inside

SetyaG replied:
Moved to its own folder

SetyaG committed:
Merge branch 'develop' into feature/PFSP1-0004

SetyaG committed:
Update ProfileCoordinator
Changed EditProfileCoordinatorProtocol into ProfileCoordinatorConstant
Changed ProfileLandingCoordinatorProtocol into ProfileCoordinatorProtocol

Update ProfileLandingController, ProfileLandingProtocol, ProfileLandingViewModel

Update CustomTableViewCell name into ProfileLandingCell

Added "call" image asset

SetyaG committed:
Deleted EditProfileCoordinator

Updated ProfileNetwork

Added ProfileModel, ProfileResponse, ProfilePath, ProfileService, ProfileSubPath

inxudianz commented on MainApp/MainApp/FeatureNetwork/ProfileNetwork/ProfileNetwork.swift:
```diff
@@ -8,6 +8,68 @@
+    func profileGet()
+}
+
+protocol EditProfileNetworkProtocol {
```
combine profileLanding and editProfile to ProfileNetwork

SetyaG replied:
combined

inxudianz commented on MainApp/MainApp/Feature/Profile/ProfileLanding/ProfileLandingController.swift:
```diff
@@ -11,20 +11,29 @@ import Component
 
 class ProfileLandingController: MasterViewController, ProfileLandingViewProtocol, UITableViewDelegate, UITableViewDataSource {
```
datasource should be handled outside of view controller

SetyaG replied:
datasource moved to view model

SetyaG committed:
Combines ProfileLanding and EditProfile's NetworkDelegate and NetworkProtocol

SetyaG committed:
Updated ProfileLandingController and ProfileLandingViewModel to separate the tableview datasource
Created a new ProfileTableViewSection file as the sections for the table view

Put a function to save edited profile on EditProfileViewModel

SetyaG committed:
Merge branch 'develop' into feature/PFSP1-0004
